packages/selectInput/components/SelectInput.tsx

```diff
@@ -0,0 +1,181 @@
+}
+
+export interface SelectInputState {
+  hasFocus: boolean;
+}
+
+class SelectInput extends React.PureComponent<
+  SelectInputProps,
+  SelectInputState
+> {
+  public static defaultProps: Partial<SelectInputProps> = {
+    appearance: SelectInputAppearance.Standard,
+    showInputLabel: true
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.handleFocus = this.handleFocus.bind(this);
+    this.handleBlur = this.handleBlur.bind(this);
+
+    this.state = {
+      hasFocus: false
+    };
+  }
+
+  public render() {
+    const {
+      appearance,
+      errors,
+      hintContent,
+      id,
+      inputLabel,
+      options,
+      showInputLabel,
+      ...other
+    } = this.props;
+    delete other.onFocus;
+    delete other.onBlur;
+
+    const hasError = this.props.appearance === SelectInputAppearance.Error;
+
+    return (
+      <FormFieldWrapper id={id} errors={errors} hintContent={hintContent}>
+        {({
+          getValidationErrors,
+          isValid,
+          errorIds,
+          hintContentId,
+          getHintContent
+        }) => (
+          <div>
+            <label
+              className={cx(getLabelStyle(hasError), {
+                [visuallyHidden]: !showInputLabel
+              })}
+              htmlFor={id}
+            >
+              {inputLabel}
+            </label>
+            <span
+              className={cx(
+                selectContainer,
+                inputContainer,
+                inputAppearances[this.getInputAppearance()],
+                display("flex")
+              )}
+            >
+              <select
+                className={cx(inputReset, select, display("block"))}
+                aria-invalid={!isValid}
+                aria-describedby={this.getDescribedBy(hintContentId, errorIds)}
+                id={id}
+                onFocus={this.handleFocus}
+                onBlur={this.handleBlur}
+                {...other}
+              >
+                {options.map((option, key) => (
+                  /* tslint:disable react-a11y-role-has-required-aria-props */
+                  /* <option> tag doesn't need additional aria markup */
+                  <option
+                    key={key}
+                    value={option.value}
+                    disabled={option.disabled}
+                  >
+                    {option.label}
+                  </option>
+                  /* tslint:enable */
+                ))}
+              </select>
+              <span className={cx(selectIcon, padding("horiz", "m"))}>
+                <Icon
+                  shape={SystemIcons.TriangleDown}
+                  size={iconSizeXs}
+                  color="inherit"
+                />
+              </span>
+            </span>
+            {getHintContent}
+            {appearance !== SelectInputAppearance.Error && getValidationErrors}
+          </div>
+        )}
+      </FormFieldWrapper>
+    );
+  }
+
+  private handleFocus(e) {
```

Comment:
I really don't like that this component has to have state just for focus styles...

I may look into detecting `:focus-within` support, and if the browser doesn't have it, don't set `outline: 0` on the input. I won't lose any sleep if somebody using Edge/IE doesn't get a pretty focus state :x